providers/aws/resources/aws.lr

```diff
@@ -3014,6 +3014,8 @@ private aws.ec2.instance @defaults("instanceId region state instanceType archite
   instanceType string
   // Tags on the instance
   tags map[string]string
+  // Instance profile of the instance.
+  IamInstanceProfile() dict
```

Comment:
Shouldn't this be `iamInstanceProfile` per our naming?

Reply:
Thanks @tas50 ! 
I updated the resource accordingly.
Works now as expected.

![instanceProfileResource](https://github.com/user-attachments/assets/2e64fa2f-bdfa-4ccb-955e-07c325786368)